server/connector.go

```diff
@@ -66,7 +67,45 @@ func (p *PlayerInfo) String() string {
 	return fmt.Sprintf("%s/%s", p.Name, p.Uuid)
 }
 
+type ServerMetrics struct {
+	sync.RWMutex
+	activeConnections map[string]int
+}
+
+func (sm *ServerMetrics) IncrementActiveConnections(serverAddress string) {
+	sm.Lock()
+	defer sm.Unlock()
+	if _, ok := sm.activeConnections[serverAddress]; !ok {
+		sm.activeConnections[serverAddress] = 1
+		return
+	}
+	sm.activeConnections[serverAddress] += 1
+}
+
+func (sm *ServerMetrics) DecrementActiveConnections(serverAddress string) {
+	sm.Lock()
+	defer sm.Unlock()
+	if activeConnections, ok := sm.activeConnections[serverAddress]; ok && activeConnections <= 0 {
+		sm.activeConnections[serverAddress] = 0
+		return
+	}
+	sm.activeConnections[serverAddress] -= 1
+}
+
+func (sm *ServerMetrics) ActiveConnectionsValue(serverAddress string) int {
+	sm.Lock()
+	defer sm.Unlock()
+	if activeConnections, ok := sm.activeConnections[serverAddress]; ok {
+		return activeConnections
+	}
+	return 0
+}
+
 func NewConnector(metrics *ConnectorMetrics, sendProxyProto bool, receiveProxyProto bool, trustedProxyNets []*net.IPNet, recordLogins bool, autoScaleUpAllowDenyConfig *AllowDenyConfig) *Connector {
+	serverMetrics := &ServerMetrics{
+		activeConnections: make(map[string]int),
+	}
```

Comment:
For consistency, introduce a `NewServerMetrics` "constructor" to wrap up the field init.